synapse_data_warehouse/synapse/dynamic_tables/V2.23.0__accessrequirement_latest.sql

```diff
@@ -0,0 +1,26 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE DYNAMIC TABLE IF NOT EXISTS ACCESSREQUIREMENT_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+    WITH latest_unique_rows AS (
+        SELECT
+            accessrequirementsnapshots.*,
+        FROM
+            {{database_name}}.synapse_raw.accessrequirementsnapshots --noqa: TMP
+        WHERE
+            CHANGE_TYPE != 'DELETE'
+        AND
+            SNAPSHOT_TIMESTAMP >= CURRENT_TIMESTAMP - INTERVAL '14 DAYS'
```

Comment:
Snapshots are taken at least every 2 weeks?

Reply:
Edit: With the update [here](https://github.com/Sage-Bionetworks/snowflake/pull/84/commits/b204445e1f264c769afd6974b169231b42724522) I do see the same 3,803 assets using 7 days as using 14 days, but I'm just going to keep it at 8 days because the time window doesn't really matter in this query.

Reply:
Excellent. Do you know if there's documentation (from platform) about what is snapshotted and how frequently? Or is everything snapshotted every 7 days?

Reply:
Btw, platform themselves use 14 days in the de-duplication.  That is probably safer because we don't want to risk snapshotting not working.

We should ask them what the optimal interval is when de-dupping.

Reply:
> what is snapshotted and how frequently?

From a conversation with Marco and from [this documentation](https://sagebionetworks.jira.com/wiki/spaces/DW/pages/3034775570/Raw+tables+in+Athena+Snowflake#Important-terminology), it looks like snapshots happen:
1. every 7 days
2. every time an interaction with an asset happens, such that its "state" changes (e.g. an account verification submission is approved, an ACL configuration is deleted, etc)

> Btw, platform themselves use 14 days in the de-duplication.

Platform team does de-duplication before uploading the data to S3?

I'm fine with keeping it at 14 days if that's what platform is using, and focusing our energies on getting the "DELETE" value in some of these snapshots tables where they're missing, since this time window method is a temporary bandaid anyway

Reply:
So when platform was doing analytics, they were doing 14 days on de-duplication. (They do not de-duplicate themselves, but when they query athena, they have to de-duplicate as well)

Reply:
@jaymedina Isn't that documentation written by [Tom and yourself](https://sagebionetworks.jira.com/wiki/spaces/DW/history/3034775570/Raw+tables+in+Athena+Snowflake)? I'll ask platform whether they have it in their own documentation.

Reply:
Yep, I assume that part of the docs came from Tom's conversations with platform but double-checking sounds good. Thanks!